charmbox-setup.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+set -ex
+
+# Bundletester and friends in tims ppa
+sudo add-apt-repository -y ppa:tvansteenburgh/ppa
+
+sudo apt-get update -qqy
+sudo apt-get install -qy  \
+                     build-essential \
+                     juju-deployer \
+                     libssl-dev \
+                     make \
+                     python-dev \
+                     python-jujuclient \
+                     python3-dev \
+                     python-pip \
+                     python3-pip \
+                     python-virtualenv \
+                     rsync  \
+                     unzip
+
+sudo apt install --no-install-recommends charm
+
+sudo pip install --upgrade pip six
+sudo pip install amulet flake8 bundletester tox
+sudo pip3 install --upgrade pip
+sudo pip3 install amulet
+
+# Install charm-tools from source
+git clone https://github.com/juju/charm-tools /tmp/charm-tools
+cd /tmp/charm-tools
+sudo pip2 install .
+cd ..
+rm -rf charm-tools
+
+# Fix for CI choking on duplicate hosts if the host key has changed
```

Comment:
@mbruzek i think we may want to move this block to jujubox, just due to the nature of what this is patching and its purpose. LMK what you think about that.